libr/bin/format/elf/elf.c

```diff
@@ -2371,7 +2371,10 @@ struct Elf_(r_bin_elf_obj_t)* Elf_(r_bin_elf_new_buf)(RBuffer *buf) {
 	struct Elf_(r_bin_elf_obj_t) *bin = R_NEW0 (struct Elf_(r_bin_elf_obj_t));
 	bin->kv = sdb_new0 ();
 	bin->b = r_buf_new ();
-	bin->size = buf->length;
+//	bin->size = buf->length;
+	ut64 curr = lseek (buf->fd, 0, SEEK_CUR);
```

Comment:
can you squash the commit with just the changes of int->ut64 to see if that breaks something? i would put all that rbuf mangling in a separate pr, to ease the review of broken tests.

this should be solved by adding an api in RBuffer to get the size instead of directly accesing the ->length field. rbuffer can do the lseek in case fd != -1 or return length otherwise, etc..

> On 01 Jul 2016, at 20:45, Rakholiya Jenish <notifications@github.com> wrote:
> 
> In libr/bin/format/elf/elf.c <https://github.com/radare/radare2/pull/5242#discussion_r69340911>:
> 
> > @@ -2371,7 +2371,10 @@ struct Elf_(r_bin_elf_obj_t)* Elf_(r_bin_elf_new_buf)(RBuffer *buf) {
> >  	struct Elf_(r_bin_elf_obj_t) *bin = R_NEW0 (struct Elf_(r_bin_elf_obj_t));
> >  	bin->kv = sdb_new0 ();
> >  	bin->b = r_buf_new ();
> > -	bin->size = buf->length;
> > +//	bin->size = buf->length;
> > +	ut64 curr = lseek (buf->fd, 0, SEEK_CUR);
> Agreed that RBuffer can work without an fd, but in this case, it is called from http://computers.tutsplus.com/tutorials/how-to-ocr-documents-for-free-in-google-drive--cms-20460 <http://computers.tutsplus.com/tutorials/how-to-ocr-documents-for-free-in-google-drive--cms-20460> which sets this. But as you commented there that we should have independent working of rbuff, in that case, it seems like the current way I am trying to implement won't be doable without the fd since I am trying to do seek+read in the r_buf_fcpy_at part to which only the RBuffer is passed and no other related info about the bin (esp. file size and fd to read from).
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/radare/radare2/pull/5242/files/171fc713a2aef8a849b4282674b931495ceb5380..3b03064a0cfe399906f78d9694b96dce5662213d#r69340911>, or mute the thread <https://github.com/notifications/unsubscribe/AA3-lqEhsBwnCE8ONFdEAEho-DQl74Cwks5qRWA0gaJpZM4JB-GF>.
>